### PR TITLE
TETP-160 Admin UI fixes

### PR DESCRIPTION
**Pull request by pekkvil**

### CODE REVIEW

File: frontend/tet/admin/src/pages/editstatic.tsx [hunk 1]
```diff
@@ -14,9 +14,10 @@ import PageNotFound from 'shared/components/pages/PageNotFound';
 import withAuth from 'shared/components/hocs/withAuth';
 import EditorLoadingError from 'tet/admin/components/editor/EditorLoadingError';
 import useLanguageOptions from 'tet-shared/hooks/translation/useLanguageOptions';
+import { Language } from 'shared/i18n/i18n';
 
 const EditStaticPage: NextPage = () => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const router = useRouter();
```
Comment (nenonja) [inline frontend/tet/admin/src/pages/editstatic.tsx:21]: You could use this hook from the shared code:
```
import useLocale from 'shared/hooks/useLocale';
...
const locale = useLocale();
...
const posting = eventToTetPosting(data, keywordResult.getKeywordType, languageOptions,
 locale);
```
It defaults to default language ('fi') when it's missing from the router context. Then you proably dont need helper function inside eventToTetPosting

Comment (nenonja) [reply frontend/tet/admin/src/pages/editstatic.tsx:21]: Nwo when I rethink this one more time, you could probably convert `eventToTetPosting` method to hook, `useEventToTetPosting` then you can use `t` and useTranslation inside and dont have to care about the localization. 

  import { useTranslation } from 'next-i18next';
 const useEventToTetPosting = (...) => 
  const { t } = useTranslation();
  return {
     name: t(event.location.name)
   ...
  }
  export default useEventTestPosting
  ```

**Review by nenonja — CHANGES_REQUESTED** [review]
some comments

File: frontend/tet/shared/src/backend-api/transformations.ts [hunk 2]
```diff
@@ -7,9 +7,20 @@ import {
 import TetPosting, { TetPostings } from 'tet-shared/types/tetposting';
 import { KeywordFn, ClassificationType } from 'tet-shared/types/keywords';
 import { OptionType } from 'tet-shared/types/classification';
+import { Language } from 'shared/i18n/i18n';
 
-export const getLocalizedString = (obj: LocalizedObject | undefined): string =>
-  obj ? obj.fi : '';
+export const getLocalizedString = (
```
[the remark this answers is not in the document]
Comment (nenonja) [reply frontend/tet/shared/src/backend-api/transformations.ts:12]: see next [comment](https://github.com/City-of-Helsinki/yjdh/pull/941#discussion_r834071022)

File: frontend/tet/admin/src/components/editor/form/ActionButtons.tsx [hunk 3]
```diff
@@ -10,18 +10,17 @@ import { useTheme } from 'styled-components';
 import useDeleteTetPosting from 'tet/admin/hooks/backend/useDeleteTetPosting';
 import { PreviewContext } from 'tet/admin/store/PreviewContext';
 import TetPosting from 'tet-shared/types/tetposting';
-import usePublishTetPosting from 'tet/admin/hooks/backend/usePublishTetPosting';
```
Comment (mattikl) [inline frontend/tet/admin/src/components/editor/form/ActionButtons.tsx:13]: TETP-180 will implement publishing as discussed at Monday's weekly, so I think we should revert these changes. ActionButtons should still have publish, but it should save the event instead of marking the saved event published. @pekkvil Let's discuss more tomorrow.

File: frontend/tet/admin/src/pages/editstatic.tsx (hunk 1 above)
Comment (pekkvil) [reply frontend/tet/admin/src/pages/editstatic.tsx:20]: Good suggestion. Changed transfromations to a single hook

File: frontend/tet/admin/src/components/editor/previewWrapper/PreviewWrapper.tsx [hunk 4]
```diff
@@ -10,35 +10,68 @@ import Container from 'tet/shared/src/components/container/Container';
...
+      if (isConfirmed) {
+        publishTetPosting.mutate(posting);
+      }
+    }
+  };
+
   return (
     <>
-      <PreviewBar />
+      <PreviewBar allowPublish={allowPublish} onPublish={publishPostingHandler} />
```
[the remark this answers is not in the document]
Comment (pekkvil) [reply frontend/tet/admin/src/components/editor/previewWrapper/PreviewWrapper.tsx:72]: Good changes for this done in TETP-180
